### PR TITLE
[Feature] Native Pay Button UI/Script for iOS

### PR DESCRIPTION
**Pull request by sleroux**
Adds a `NativePayButtonUI` script and associated native plugin code for IOS to generate the Apple Pay button from system frameworks.

### CODE REVIEW

File: scripts/generator/graphql_generator/csharp/Editor/BuildPipeline/iOSPostProcessor.cs.erb [hunk 1]
```diff
@@ -67,6 +68,18 @@ namespace <%= namespace %>.Editor.BuildPipeline {
                 File.WriteAllLines (swiftInterfaceHeaderPath, lines);
             }
         }
+
+        /// Sets the correct target for Shopify Tests
+        private static void SetCorrectTestsTarget(ExtendedPBXProject project) {
```
Comment (sleroux) [inline scripts/generator/graphql_generator/csharp/Editor/BuildPipeline/iOSPostProcessor.cs.erb]: Annoying issue when exporting the example app for running on iOS

**Comment by sleroux** [comment]
Snapshot from adding it to the example app (not in this branch)
img_4312

**Comment by dengjeffrey** [comment]
😮 🎉

File: scripts/generator/graphql_generator/csharp/UI/NativePayButtonUI.cs.erb [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    using System.Runtime.InteropServices;
+    using System.Text;
+    using System;
+
+    /// <summary>
+    /// The Apple Pay payment button style C# enum. See https://developer.apple.com/documentation/passkit/pkpaymentbuttonstyle
+    /// for all available options.
+    /// </summary>
+    public enum ApplePayButtonStyle {
+        white,
```
Comment (dengjeffrey) [inline scripts/generator/graphql_generator/csharp/UI/NativePayButtonUI.cs.erb]: The convention is to use `Caps` case for enums in `C#`, Up to you 🤷‍♂️, but I think the rest of the Unity SDK is using this convention

File: scripts/generator/graphql_generator/csharp/UI/NativePayButtonUI.cs.erb [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        /// </summary>
+        public ApplePayButtonType applePayButtonType;
+
+        /// <summary>
+        /// When set to true, the the Apple Pay button image will include the 10% minimum spacing around the 
+        /// button image required by Apple's design guidelines.
+        /// </summary>
+        public bool includeMinimumSpacingForIOS;
+
+        [DllImport ("__Internal")]
```
Comment (dengjeffrey) [inline scripts/generator/graphql_generator/csharp/UI/NativePayButtonUI.cs.erb]: This doesn't cause a problem when exporting to Android?

File: Assets/Shopify/Plugins/iOS/Shopify/ApplePayButtonGenerator.mm [hunk 4]
```diff
@@ -0,0 +1,89 @@
+#import <Foundation/Foundation.h>
```
Comment (dengjeffrey) [inline Assets/Shopify/Plugins/iOS/Shopify/ApplePayButtonGenerator.mm]: Don't think this one is needed, `UIKit` imports `Foundation`

File: Assets/Shopify/Plugins/iOS/Shopify/ApplePayButtonGenerator.mm [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+#import <UIKit/UIKit.h>
+#import <PassKit/PassKit.h>
+#import "UnityAppController.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+    PKPaymentButtonStyle SBPayButtonStyleFromString(const char *style) {
+        if (strcmp(style, "white") == 0) {
```
[the remark this answers is not in the document]
Comment (sleroux) [reply Assets/Shopify/Plugins/iOS/Shopify/ApplePayButtonGenerator.mm]: Good catch 👍 I'll update to use `strncmp`

**Comment by sleroux** [comment]
@dengjeffrey Fixed up review comments and cleaned up the platform ifdefs in `NativePayButtonUI`.

File: Assets/Shopify/Plugins/iOS/Shopify/ApplePayButtonGenerator.mm [hunk 6]
```diff
@@ -7,41 +6,40 @@
...
+        if (strncmp(type, "PLAIN", 5) == 0) {
             return PKPaymentButtonTypePlain;
-        } else if (strcmp(type, "buy") == 0) {
+        } else if (strncmp(type, "BUY", 3) == 0) {
             return PKPaymentButtonTypeBuy;
-        } else if (strcmp(type, "setUp") == 0) {
+        } else if (strncmp(type, "SETUP", 5) == 0) {
             return PKPaymentButtonTypeSetUp;
-        } else if (strcmp(type, "inStore") == 0) {
+        } else if (strncmp(type, "IN_STORE", 6) == 0) {
```
Comment (dengjeffrey) [inline Assets/Shopify/Plugins/iOS/Shopify/ApplePayButtonGenerator.mm]: Should this be a length of 7? This seems kind of error prone, perhaps the constants could be declared like so?
```C
char plainType[] = "PLAIN";
strncmp(type, plainType, sizeof(plain));
```

**Comment by sleroux** [comment]
@dengjeffrey Sorry for the delay - should be up to date now with latest fixes.

**Comment by sleroux** [comment]
Ignoring Android tests for now (blocked on lint issues that are on beta).

**Pull request merged**